YoutubeDownloader/ViewModels/Framework/ViewModelFactory.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using YoutubeDownloader.ViewModels.Components;
+using YoutubeDownloader.ViewModels.Dialogs;
+
+namespace YoutubeDownloader.ViewModels.Framework;
+
+public class ViewModelFactory(IServiceProvider serviceProvider) : IViewModelFactory
+{
+    public DashboardViewModel CreateDashboardViewModel() => GetViewModel<DashboardViewModel>();
+
+    public DownloadMultipleSetupViewModel CreateDownloadMultipleSetupViewModel() =>
+        GetViewModel<DownloadMultipleSetupViewModel>();
+
+    public AuthSetupViewModel CreateAuthSetupViewModel() => GetViewModel<AuthSetupViewModel>();
+
+    public DownloadSingleSetupViewModel CreateDownloadSingleSetupViewModel() =>
+        GetViewModel<DownloadSingleSetupViewModel>();
+
+    public DownloadViewModel CreateDownloadViewModel() => GetViewModel<DownloadViewModel>();
+
+    public MessageBoxViewModel CreateMessageBoxViewModel() => GetViewModel<MessageBoxViewModel>();
+
+    public SettingsViewModel CreateSettingsViewModel() => GetViewModel<SettingsViewModel>();
+
+    private T GetViewModel<T>() => ActivatorUtilities.CreateInstance<T>(serviceProvider);
```

Comment:
What's the difference between this and `serviceProvider.GetRequiredService<...>()`?

Reply:
`serviceProvider.GetRequiredService<...>()` only resolves services that were registered in the DI container, when using ActivatorUtilities it can create a new instance and resolving it's parameters from DI without it beeing registered

Reply:
Ahhh I see. Would it get it from the provider if it _is_ registered? Or would it still create a new instance regardless?

Reply:
it will always create a new instance the described way, but there is also a `GetServiceOrCreateInstance` method that does what you asked. I think it would make sense to change to it, so if later you decide to register a ViewModel, it would use that registration